File: src/main/java/com/abysalto/middleware/model/Role.java
package com.abysalto.middleware.model;

/**
 * Application roles. Persisted by name ({@code @Enumerated(STRING)}) and exposed to Spring Security
 * as {@code ROLE_<name>} authorities. Only a single role exists today; the enum keeps role handling
 * type-safe and gives an obvious place to grow.
 */
public enum Role {
    USER;

    /** Spring Security authority form of this role (e.g. {@code ROLE_USER}). */
    public String authority() {
        return "ROLE_" + name();
    }
}

File: dotnet/src/Middleware.Core/Domain/Role.cs
namespace Middleware.Core.Domain;

/// <summary>
/// Application roles. Persisted by name and exposed to authorization as <c>ROLE_&lt;name&gt;</c>
/// authorities. Only a single role exists today; the enum keeps role handling type-safe.
/// </summary>
public enum Role
{
    User
}

/// <summary>Helpers for <see cref="Role"/>.</summary>
public static class RoleExtensions
{
    /// <summary>Authority form of this role (e.g. <c>ROLE_USER</c>), matching the Java service.</summary>
    public static string Authority(this Role role) => "ROLE_" + role.ToString().ToUpperInvariant();
}
